blocklist-client/src/client/risk_client.rs

```diff
@@ -0,0 +1,352 @@
+use crate::common::error::Error;
+use crate::common::{BlocklistStatus, RiskSeverity};
+use crate::config::RiskAnalysisConfig;
+use reqwest::{Client, Response, StatusCode};
+use serde::Deserialize;
+use tracing::debug;
+const API_BASE_PATH: &str = "/api/risk/v2/entities";
+
+#[derive(Deserialize, Debug)]
+struct RegistrationResponse {
+    address: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct RiskResponse {
+    risk: Option<String>,
+    #[serde(rename = "riskReason")]
+    risk_reason: Option<String>,
+}
```

Comment:
Why is `risk` optional here? The `get_risk_assessment` function returns an `InvalidApiResponse` error if this is not provided.

Also, it seems to me that this type is only an intermediate representation used to construct a `RiskAssessment`. It would be simpler to try to deserialize the response in `get_risk_assessment` directly as a `RiskAssessment`, and skip the intermediate `RiskResponse` struct completely. Am I missing something here or is there any good reason to keep the `RiskResponse` type around?

Reply:
Removed optional for risk (but risk_reason needs to be optional)
I wanted to separate the API data mapping from the business logic struct...in case we want to manipulate those differently but since this is not the case here, I removed the RiskResponse type -> this made the error mapping a bit different